pyogrio/tests/test_geopandas_io.py

```diff
@@ -1010,67 +1020,96 @@ def test_write_read_empty_dataframe_unsupported(tmp_path, ext):
         _ = read_dataframe(filename)
 
 
-def test_write_dataframe_gpkg_multiple_layers(tmp_path, naturalearth_lowres):
+def test_write_dataframe_gpkg_multiple_layers(tmp_path, naturalearth_lowres, use_arrow):
     input_gdf = read_dataframe(naturalearth_lowres)
     output_path = tmp_path / "test.gpkg"
 
-    write_dataframe(input_gdf, output_path, layer="first", promote_to_multi=True)
+    write_dataframe(
+        input_gdf,
+        output_path,
+        layer="first",
+        promote_to_multi=True,
+        use_arrow=use_arrow,
+    )
 
     assert os.path.exists(output_path)
     assert np.array_equal(list_layers(output_path), [["first", "MultiPolygon"]])
 
-    write_dataframe(input_gdf, output_path, layer="second", promote_to_multi=True)
+    write_dataframe(
+        input_gdf,
+        output_path,
+        layer="second",
+        promote_to_multi=True,
+        use_arrow=use_arrow,
+    )
     assert np.array_equal(
         list_layers(output_path),
         [["first", "MultiPolygon"], ["second", "MultiPolygon"]],
     )
 
 
 @pytest.mark.parametrize("ext", ALL_EXTS)
-def test_write_dataframe_append(tmp_path, naturalearth_lowres, ext):
+def test_write_dataframe_append(request, tmp_path, naturalearth_lowres, ext, use_arrow):
     if ext == ".fgb" and __gdal_version__ <= (3, 5, 0):
         pytest.skip("Append to FlatGeobuf fails for GDAL <= 3.5.0")
 
     if ext in (".geojsonl", ".geojsons") and __gdal_version__ <= (3, 6, 0):
         pytest.skip("Append to GeoJSONSeq only available for GDAL >= 3.6.0")
 
+    if use_arrow and ext.startswith(".geojson"):
+        request.node.add_marker(
+            pytest.mark.xfail(reason="Bugs with append when writing Arrow to GeoJSON")
```

Comment:
Is there a GDAL issue or version range to note in a comment here?

Reply:
Opened https://github.com/OSGeo/gdal/issues/9792 (will add that link in the comments here)